File: src/brs/db/store/DigitalGoodsStoreStore.java
package brs.db.store;

import brs.DigitalGoodsStore;
import brs.crypto.EncryptedData;
import brs.db.SignumKey;
import brs.db.VersionedEntityTable;
import brs.db.VersionedValuesTable;

import java.util.Collection;

public interface DigitalGoodsStoreStore {

  SignumKey.LongKeyFactory<DigitalGoodsStore.Purchase> getFeedbackDbKeyFactory();

  SignumKey.LongKeyFactory<DigitalGoodsStore.Purchase> getPurchaseDbKeyFactory();

  VersionedEntityTable<DigitalGoodsStore.Purchase> getPurchaseTable();

  VersionedValuesTable<DigitalGoodsStore.Purchase, EncryptedData> getFeedbackTable();

  SignumKey.LongKeyFactory<DigitalGoodsStore.Purchase> getPublicFeedbackDbKeyFactory();

  VersionedValuesTable<DigitalGoodsStore.Purchase, String> getPublicFeedbackTable();

  SignumKey.LongKeyFactory<DigitalGoodsStore.Goods> getGoodsDbKeyFactory();

  VersionedEntityTable<DigitalGoodsStore.Goods> getGoodsTable();

  Collection<DigitalGoodsStore.Goods> getGoodsInStock(int from, int to);

  Collection<DigitalGoodsStore.Goods> getSellerGoods(long sellerId, boolean inStockOnly, int from, int to);

  Collection<DigitalGoodsStore.Purchase> getAllPurchases(int from, int to);

  Collection<DigitalGoodsStore.Purchase> getSellerPurchases(long sellerId, int from, int to);

  Collection<DigitalGoodsStore.Purchase> getBuyerPurchases(long buyerId, int from, int to);

  Collection<DigitalGoodsStore.Purchase> getSellerBuyerPurchases(long sellerId, long buyerId, int from, int to);

  Collection<DigitalGoodsStore.Purchase> getPendingSellerPurchases(long sellerId, int from, int to);

  Collection<DigitalGoodsStore.Purchase> getExpiredPendingPurchases(int timestamp);
}

File: src/main/kotlin/brs/db/store/DigitalGoodsStoreStore.kt
package brs.db.store

import brs.DigitalGoodsStore
import brs.crypto.EncryptedData
import brs.db.BurstKey
import brs.db.VersionedEntityTable
import brs.db.VersionedValuesTable

interface DigitalGoodsStoreStore {

    val feedbackDbKeyFactory: BurstKey.LongKeyFactory<DigitalGoodsStore.Purchase>

    val purchaseDbKeyFactory: BurstKey.LongKeyFactory<DigitalGoodsStore.Purchase>

    val purchaseTable: VersionedEntityTable<DigitalGoodsStore.Purchase>

    val feedbackTable: VersionedValuesTable<DigitalGoodsStore.Purchase, EncryptedData>

    val publicFeedbackDbKeyFactory: BurstKey.LongKeyFactory<DigitalGoodsStore.Purchase>

    val publicFeedbackTable: VersionedValuesTable<DigitalGoodsStore.Purchase, String>

    val goodsDbKeyFactory: BurstKey.LongKeyFactory<DigitalGoodsStore.Goods>

    val goodsTable: VersionedEntityTable<DigitalGoodsStore.Goods>

    fun getGoodsInStock(from: Int, to: Int): Collection<DigitalGoodsStore.Goods>

    fun getSellerGoods(sellerId: Long, inStockOnly: Boolean, from: Int, to: Int): Collection<DigitalGoodsStore.Goods>

    fun getAllPurchases(from: Int, to: Int): Collection<DigitalGoodsStore.Purchase>

    fun getSellerPurchases(sellerId: Long, from: Int, to: Int): Collection<DigitalGoodsStore.Purchase>

    fun getBuyerPurchases(buyerId: Long, from: Int, to: Int): Collection<DigitalGoodsStore.Purchase>

    fun getSellerBuyerPurchases(sellerId: Long, buyerId: Long, from: Int, to: Int): Collection<DigitalGoodsStore.Purchase>

    fun getPendingSellerPurchases(sellerId: Long, from: Int, to: Int): Collection<DigitalGoodsStore.Purchase>

    fun getExpiredPendingPurchases(timestamp: Int): Collection<DigitalGoodsStore.Purchase>
}
